util/logger/src/lib.rs

```diff
@@ -174,3 +177,32 @@ pub fn init(config: Config) -> Result<LoggerInitGuard, SetLoggerError> {
 pub fn flush() {
     log::logger().flush()
 }
+
+fn setup_panic_logger() {
+    let next = panic::take_hook();
+    let panic_logger = |info: &panic::PanicInfo| {
+        let backtrace = Backtrace::new();
+        let thread = thread::current();
+        let name = thread.name().unwrap_or("unnamed");
+        let location = info.location().unwrap(); // The current implementation always returns Some
+        let msg = match info.payload().downcast_ref::<&'static str>() {
+            Some(s) => *s,
+            None => match info.payload().downcast_ref::<String>() {
+                Some(s) => &**s,
+                None => "Box<Any>",
+            },
+        };
+        log::error!(
+            target: "panic", "thread '{}' panicked at '{}': {}:{}{:?}",
+            name,
+            msg,
+            location.file(),
+            location.line(),
+            backtrace,
+        );
+    };
+    panic::set_hook(Box::new(move |info| {
+        panic_logger(info);
+        next(info);
```

Comment:
We don't want the default hook. This will print the panic twice. Just override the default hook, and add a comment in `Setup` that logger must init first before other panic hooks.

Reply:
Updated:

* Remove default hook
* Add comments in `Setup::setup_app` and `setup_panic_logger`